relays/client-substrate/src/chain.rs

```diff
@@ -54,14 +54,16 @@ pub trait Chain: ChainBase + Clone {
 	type SignedBlock: Member + Serialize + DeserializeOwned + BlockWithJustification<Self::Header>;
 	/// The aggregated `Call` type.
 	type Call: Dispatchable + Debug;
+	/// Balance of an account in native tokens.
+	///
+	/// The chain may suport multiple tokens, but this particular type is for token that is used
+	/// to pay for transaction dispatch, to reward different relayers (headers, messages), etc.
+	type Balance: Parameter + Member + DeserializeOwned + Clone + Copy + CheckedSub + PartialOrd + Zero;
```

Comment:
Does it have to be the native token? Would it work if there are multiple instances of `balances` pallet?

Reply:
Honestly - I'm not sure what's the difference between `TOKEN1` and `TOKEN2` (different instances of the balance/assets pallet). Which token is native then? :) The type I'm talking here is the type that is used to pay transaction fees, relayer rewards. Depending on the runtime implementation, this may be any instance of the balance pallet. But what this PR does, is that it brings knowledge that all payments that he may see during its work will be conducted in the token that is defined here.

Reply:
So why it was not required before - because relayer was just ignoring fees and rewards. Greedy relayer would need to know that => that's why I've moved that in a base trait.

Reply:
So maybe we shouldn't be calling them "native tokens", and instead call them "payment tokens".

It's totally reasonable for a runtime to have many instances for the balance pallet, and perhaps accept any of those tokens as a way to pay fees.